pkg/profiling/profiling_test.go

```diff
@@ -0,0 +1,79 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+func Test_fifoDir(t *testing.T) {
+	dir := t.TempDir()
+	maxFileCount := 3
+	fs := fifoDir{dirPath: dir, maxFileCount: maxFileCount}
+
+	// This should be bigger than maxFileCount to test FIFO deletion is done properly
+	numFilesToCreate := 10
+	for i := 0; i < numFilesToCreate; i++ {
+		fs.Create(fmt.Sprintf("%d.test.dump", i))
```

Comment:
Depending on the time resolution of underlying FS, you might have to inject a tiny delay here for the test to not flake.